sycl/test/check_device_code/device_global_const_eval_use.cpp

```diff
@@ -0,0 +1,104 @@
+// RUN: %clangxx -std=c++20 -fsycl -c -fsycl-device-only -S -emit-llvm %s -o - | FileCheck %s
+
+// Tests that device_globals with device_image_scope property can be compile
```

Comment:
you are missing char, and char[], out of the fundamental types. 

You are missing a multidimensional array test as well. 

You should try it with a struct that has a `constexpr` constructor for example one that needs to calculate a sequence (ie. Fibonacci)

Reply:
Should the user expect to initialize a multi-dimensional device_global like the following:
`dg_multi_dim_arr{{3, 4}, {5, 6}, {7, 8}};`? Or is `dg_multi_dim_arr{3, 4, 5, 6, 7, 8};` the expected way?

Reply:
Yeah that is one unfortunate side effect of the initialization method, to initialize multidimensional arrays it is just one big list of values, there is no `{}` separators for each dimension (same thing for arrays of structs). I think this is acceptable.